### PR TITLE
refactor: [M3-6503] - Combine SelectPlanPanel and SelectPlanQuantityPanel components - part 1

### PR DESCRIPTION
**Pull request by cpathipa**
## Description 📝

- This PR simplifies  `SelectPlanPanel` and  `SelectPlanQuantityPanel` components.
- Removes redundant / duplicate code.
- Provides single source of truth to generate tabs and plan types from utils `getPlanSelectionsByPlanType` and  `planTabInfoContent`.
- As a back-out plan, original components are untouched, created new components (`PlansPanel` and `KubernetesPlansPanel`) and referenced in respective plans tables.

## Major Changes 🔄
- References PlansPanel instead of SelectPlanPanel
- References KubernetesPlansPanel instead of SelectPlanQuantityPanel

## Preview 📷
No visual changes. 

## How to test 🧪
- Validate Linodes Plans Table.
- Validate Database Plans Table.
- Validate Kubernetes Plans Table.
- Validate general functionalities in Linode create / clone / backup/ resize/ rebuild flow's.
- Validate general functionalities in Database create flow.
- Validate general functionalities in Kubernetes create cluster flow.
- yarn test 
- yarn cy:run

### CODE REVIEW

**Comment by bnussman** [comment]
Please add a chageset to your PR 🥺🚨

**Comment by jaalah-akamai** [comment]
@cpathipa I think the description got deleted?

**Review by dwiley-akamai — CHANGES_REQUESTED** [review]
The Linode Create and Kubernetes Create pages crash for me (as well as the Resize modal):

Screenshot 2023-06-08 at 10 23 29 AM

Screenshot 2023-06-08 at 10 24 02 AM

File: packages/manager/src/features/Linodes/LinodesCreate/SelectPlanPanel/utils.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      plansByType[type.class].push(type);
+    }
+  }
+
+  // filter empty plan group
+  Object.keys(plansByType).forEach((key) => {
+    if (plansByType[key].length === 0) {
+      delete plansByType[key];
+    }
+  });
```
Comment (jaalah-akamai) [inline packages/manager/src/features/Linodes/LinodesCreate/SelectPlanPanel/utils.ts:62]: We could follow the same pattern as you did above by using `reduce` to filter the plan:
```
const filteredPlansByType = Object.keys(plansByType).reduce((acc, key) => {
    if (plansByType[key].length > 0) {
      acc[key] = plansByType[key];
    }
    return acc;
  }, {} as PlansByType<T>);

return filteredPlansByType;
```

**Comment by cpathipa** [comment]
> @cpathipa I think the description got deleted?

@jaalah-akamai its the same [PR](https://github.com/linode/manager/pull/9165). but, merging to develop.

**Comment by cpathipa** [comment]
> The Linode Create and Kubernetes Create pages crash for me (as well as the Resize modal and Add Node Pool drawer):
> 
> Screenshot 2023-06-08 at 10 23 29 AM
> 
> Screenshot 2023-06-08 at 10 24 02 AM

@dwiley-akamai handled this scenario.

File: packages/manager/src/features/Linodes/LinodesCreate/SelectPlanPanel/KubernetesPlanContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import { KubernetesPlanSelection } from './KubernetesPlanSelection';
+
+const tableCells = [
+  { cellName: 'Plan', testId: 'plan', center: false, noWrap: false },
+  { cellName: 'Monthly', testId: 'monthly', center: false, noWrap: false },
+  { cellName: 'Hourly', testId: 'hourly', center: false, noWrap: false },
+  { cellName: 'RAM', testId: 'ram', center: true, noWrap: false },
+  { cellName: 'CPUs', testId: 'cpu', center: true, noWrap: false },
+  { cellName: 'Storage', testId: 'storage', center: true, noWrap: false },
+  { cellName: 'Quantity', testId: 'quantity', center: false, noWrap: false },
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Linodes/LinodesCreate/SelectPlanPanel/KubernetesPlanContainer.tsx:19]: It looks a tad odd to not center `Quantity` IMO:

Screenshot 2023-06-08 at 12 42 59 PM

File: packages/manager/src/features/Linodes/LinodesCreate/SelectPlanPanel/utils.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+import arrayToList from 'src/utilities/arrayToDelimiterSeparatedList';
+import { PlanSelectionType } from './PlansPanel';
+
+export type PlansTypes<T> = Record<LinodeTypeClass, T[]>;
+
+type PlansByType<T> = Omit<PlansTypes<T>, 'nanode' | 'standard'> & {
+  shared: T[];
+};
+
+//We could update this to add or remove any new or existing plan tabs.
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Linodes/LinodesCreate/SelectPlanPanel/utils.ts:14]:
```suggestion
// We could update this to add or remove any new or existing plan tabs.
```

File: packages/manager/src/features/Linodes/LinodesCreate/SelectPlanPanel/utils.ts [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+  T extends { class: LinodeTypeClass }
+>(
+  types: T[]
+): PlansByType<T> => {
+  const plansByType: PlansByType<T> = planTypeOrder.reduce((acc, key) => {
+    acc[key] = [];
+    return acc;
+  }, {} as PlansByType<T>);
+
+  //group plans by type
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Linodes/LinodesCreate/SelectPlanPanel/utils.ts:48]:
```suggestion
// group plans by type
```

**Review by dwiley-akamai — APPROVED** [review]

**Review by bnussman-akamai — APPROVED** [review]
I tested general provisioning of Linodes and LKE clusters

**Pull request merged**